tools/autograd/templates/python_variable_methods.cpp

```diff
@@ -33,6 +35,84 @@ static PyObject * THPVariable_detach_(PyObject* self, PyObject* args)
   END_HANDLE_TH_ERRORS
 }
 
+static IntList dispatch_size(const Tensor& self) {
+  // avoid releasing GIL/changing device, should be quick
+  // yes, this is called sizes in ATen.
+  return self.sizes();
+}
+
+static int64_t dispatch_size(const Tensor& self, int64_t dim) {
+  // avoid releasing GIL/changing device, should be quick
+  return self.size(dim);
+}
+
+static PyObject * THPVariable_size(PyObject* self, PyObject* args, PyObject* kwargs)
+{
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({
+    "size(int64_t dim)",
+    "size()",
+  });
+  auto& self_ = reinterpret_cast<THPVariable*>(self)->cdata;
+  PyObject* parsed_args[3];
+  auto r = parser.parse(args, kwargs, parsed_args);
+  if (r.idx == 0) {
+    return wrap(dispatch_size(self_, r.toInt64(0)));
+  } else if (r.idx == 1) {
+    // we can't do the normal wrapping here because IntList maps to both
+    // torch.Size and tuple in python
+    IntList sizes = dispatch_size(self_);
+    return THPSize_New(sizes.size(), (int64_t *)sizes.data());
```

Comment:
Why the cast? That looks fishy. If it's for const-ness change THPSize_New so that it takes a const pointer.